hyperspy/signal.py

```diff
@@ -2249,6 +2251,73 @@ def rebin(self, new_shape, out=None):
 
     rebin.__doc__ %= OUT_ARG
 
+    def linear_bin(self, scale, crop=True):
+        """
+        Binning of the spectrum image by a non-integer pixel value.
+
+        Parameters
+        ----------
+        scale: a list of floats for each dimension specify the new:old pixel
+               ratio
+              e.g. [1, 1, 2]
+              a ratio of 1 is no binning in the x and y directions.
+              a ratio of 2 means that each pixel in the new spectrum is
+              twice the width of the pixels in the old spectrum, in the energy,
+              dimension.
+         crop: when binning by a non-integer number of pixels it is likely that
+               the final row in each dimension contains less than the full
+               quota to fill one pixel.
+             e.g. 5*5 array binned by 2.1 will produce two rows containing 2.1
+             pixels and one row containing only 0.8 pixels worth. Selection of
+             crop = 'True' or crop = 'False' determines whether or not this
+             'black' line is cropped from the final binned array or not.
+
+        *Please note that if crop=False is used, the final row in each
+        dimension may appear black, if a fractional number of pixels are left
+        over. It can be removed but has been left to preserve total counts
+        before and after binning.*
+
+        Return
+        ------
+        A new spectrum image with new dimensions width/scale for each
+        dimension in the data. The axes scales but not the dwell_time/exposure
+        have been corrected accordingly.
+
+        Examples
+        --------
+        Input:
+        spectrum = hs.signals.EDSTEMSpectrum(np.ones([4, 4, 10]))
```

Comment:
I guess that's the difference between using command line and notebook. Happy to follow and comparable format though.